lisa/schema.py

```diff
@@ -696,6 +697,12 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
                     "one of admin_password and admin_private_key_file must be set"
                 )
 
+        allow_list = [True, False, "no", "always", "onfailed"]
```

Comment:
Use an enum. I'm not sure if no is needed here, it should be translate to boolean by yaml lib. If not, we should add "yes" also, and make them case insensitive.